presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -1099,13 +1100,19 @@ public static double round(@SqlType(StandardTypes.DOUBLE) double num, @SqlType(S
         if (Double.isNaN(num) || Double.isInfinite(num)) {
             return num;
         }
-
         double factor = Math.pow(10, decimals);
-        if (num < 0) {
-            return -(Math.round(-num * factor) / factor);
+        double scaled = num * factor;
```

Comment:
this seems a bit brittle because 1) if num * factor is greater than max double you'll overflow the double  2) seems like you could have corner cases by using an inexact type to check the exact type size, .

Maybe use bigdecimal all the time (don't know the cost of this), or use guava's DoubleMath.roundToLong() and if it throws an arithmetic catch it and use bigdecimal.

Reply:
Thanks @rschlussel . I'll go with the DoubleMath.roundToLong() to minimize the overhead added to the original path.